float64 cloest_point
float64 farthest_point
